packages/client/src/hooks/usuAuth.ts

```diff
@@ -0,0 +1,84 @@
+import { RouterList } from './../router/routerList';
+import { useNavigate } from 'react-router-dom';
+import { authSelectors } from '../store/auth';
+import {
+  useLazyCheckAuthUserQuery,
+  useLogoutMutation,
+  useSignInMutation,
+  useSignUpMutation,
+} from '../store/auth/api';
+import { useActions } from './useActions';
+import { useTypeSelector } from './useTypeSelector';
+import type { RequestSignIn, RequestSignUp } from '../store/auth/type';
+
+export const useAuth = () => {
+  const { isAuth, isLoadingAuth } = useTypeSelector(authSelectors.allAuth);
+
+  const { setIsAuth, resetUserState, setIsLoadingAuth, setUserId } =
+    useActions();
+
+  const navigate = useNavigate();
+
+  const [fetchIsAuth] = useLazyCheckAuthUserQuery();
+  const [fetchLogout] = useLogoutMutation();
+  const [fetchSignIn] = useSignInMutation();
+  const [fetchSignUp] = useSignUpMutation();
+
+  const checkIsAuth = () => {
+    setIsLoadingAuth(true);
+
+    fetchIsAuth(null)
+      .then(res => {
+        const { isSuccess } = res;
+
+        setIsAuth(isSuccess);
```

Comment:
```suggestion
      .then(({ isSuccess }) => setIsAuth(isSuccess))
```
Лучше деструктурировать прямо в параметре. Иначе смысла немного в 32 строке.